agent/structs/config_entry_test.go

```diff
@@ -2216,3 +2216,24 @@ func requireContainsLower(t *testing.T, haystack, needle string) {
 func intPointer(i int) *int {
 	return &i
 }
+
+func TestConfigEntryQuery_CacheInfoKey(t *testing.T) {
+	assertCacheInfoKeyIsComplete(t, &ConfigEntryQuery{})
+}
+
+func TestServiceConfigRequest_CacheInfoKey(t *testing.T) {
+	ignoredFields := []string{
+		// TODO: should QueryOptions.Filter be included in the key?
+		"QueryOptions",
+		// TODO: should this be included in the key?
+		"MeshGateway",
+		// TODO: should this be included in the key?
+		"Mode",
+	}
+	assertCacheInfoKeyIsComplete(t, &ServiceConfigRequest{}, ignoredFields...)
+}
+
+func TestDiscoveryChainRequest_CacheInfoKey(t *testing.T) {
+	// TODO: should QueryOptions.Filter be included in the key?
+	assertCacheInfoKeyIsComplete(t, &DiscoveryChainRequest{}, "QueryOptions")
```

Comment:
Seems like `QueryOptions.Filter` is not used by the endpoint, so maybe this ignore is ok here.

Reply:
What is the downside to preemptively adding QueryOptions.Filter in the requests where it is available? We could start using it later and not remember to un-ignore it here. Any time it's used it will influence the results.

Reply:
No downside I can think of. I had a similar thought, but decided to leave it as a TODO to get input from others before making the change.

I realized while working on this that `assertCacheInfoKeyIsComplete` is only way of the first level struct, not any nested structs. So missed changes to fields that are structs may not be caught by these tests. I think we can improve `assertCacheInfoKeyIsComplete` to handle that, but maybe as a follow up. This test coverage should case most cases and is better than what we had before.